src/reducers/file-search.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+/**
+ * UI reducer
+ * @module reducers/ui
+ */
+
+import makeRecord from "../utils/makeRecord";
+import { prefs } from "../utils/prefs";
+
+import type { Action, panelPositionType } from "../actions/types";
+import type { Record } from "../utils/makeRecord";
+
+export type Modifiers = Record<{
+  caseSensitive: boolean,
+  wholeWord: boolean,
+  regexMatch: boolean
+}>;
+
+export type MatchedLocations = {
+  line: number,
+  ch: number
+};
+
+export type SearchResults = {
+  matches: Array<MatchedLocations>,
+  matchIndex: number,
+  index: number,
+  count: number
+};
+
+export type UIState = {
+  searchResults: SearchResults,
+  active: boolean,
+  query: string,
+  modifiers: Modifiers
+};
+
+export const State = makeRecord(
+  ({
+    active: false,
+    query: "",
+    searchResults: {
+      matches: [],
+      matchIndex: -1,
+      index: -1,
+      count: 0
+    },
+    modifiers: makeRecord({
+      caseSensitive: prefs.fileSearchCaseSensitive,
+      wholeWord: prefs.fileSearchWholeWord,
+      regexMatch: prefs.fileSearchRegexMatch
+    })()
+  }: UIState)
+);
+
+function update(
+  state: Record<UIState> = State(),
+  action: Action
+): Record<UIState> {
+  switch (action.type) {
+    case "TOGGLE": {
+      return state.set("active", action.value);
+    }
+
+    case "SET_QUERY": {
+      return state.set("query", action.query);
+    }
+
+    case "SET_SEARCH_RESULTS": {
+      return state.set("searchResults", action.results);
+    }
+
+    case "TOGGLE_MODIFIER": {
+      const actionVal = !state.getIn(["modifiers", action.modifier]);
```

Comment:
I'm on the fence about this but i will mention it anyway; is action.modifier a string? could we do something like  
```javascript
prefs[`fileSearch${action.modifier}`]
```
(i think it needs a first letter uppercase, but you get the idea)

since we seem to be doing the same thing each time, and we might not need this extra layer